File: elementary/first-word.rb
#!/usr/bin/env ruby

def first_word(text)
    return text.scan(/([a-zA-Z']+)/)[0][0]
end

if __FILE__ == $0
    puts(first_word("Hello World"))

    raise unless first_word("Hello World") == "Hello"
    raise unless first_word(" more spaces ") == "more"
    raise unless first_word("don't split it") == "don't"
    raise unless first_word("hello, world") == "hello"
    raise unless first_word("... very long sentence ...") == "very"
    raise unless first_word("mokus") == "mokus"
end

File: elementary/first-word.py
#!/usr/bin/env python3

import re

def first_word(text: str) -> str:
    return re.search("([a-zA-Z']+)", text)[1]

if __name__ == '__main__':
    print(first_word("Hello World"))

    assert first_word("Hello World") == "Hello"
    assert first_word(" more spaces ") == "more"
    assert first_word("don't split it") == "don't"
    assert first_word("hello, world") == "hello"
    assert first_word("... very long sentence ...") == "very"
    assert first_word("mokus") == "mokus"
